### PR TITLE
fix pairwise-difference no variance

### LINKED ISSUE #148: pairwise-differences: t-value referenced before assignment
**Bug Description**
the t-value in pairwise-differences is not being assigned properly when there is no variance between groups. These lines should be amended:
https://github.com/qiime2/q2-longitudinal/blob/8bf27af74a9ff43844ab860132b051098367dcf3/q2_longitudinal/_utilities.py#L241-L244

Probably to assign `t = p = np.nan` instead of `pass`

**References**
See [forum xref](https://forum.qiime2.org/t/longitudinal-pairwise-difference-error/12389/7?u=nicholas_bokulich)

### PR DESCRIPTION
**Pull request by nbokulich**
fixes #148

### CODE REVIEW

File: q2_longitudinal/_utilities.py [hunk 1]
```diff
@@ -240,7 +240,7 @@ def _compare_pairwise_differences(groups, parametric=False):
                 t, p = wilcoxon(values)
         except ValueError:
             # if test fails (e.g., because of zero variance), just skip
```
Comment (thermokarst) [inline q2_longitudinal/_utilities.py]: Drive-by Comment:

Don't forget to update or remove this comment.

Comment (nbokulich) [reply q2_longitudinal/_utilities.py]: I left in intentionally... I see assigning `np.nan` as still skipping, just in a more functional way, so I think this comment still applies but happy to update if you have a suggestion.

Comment (thermokarst) [reply q2_longitudinal/_utilities.py]: Oh I think I see now, although that wasn't clear to me at first. How about restructuring a bit? Revert back to the `pass` here (keep the comment as-is), and move the `t = p = np.nan` business up just before the `try` statement. This proposed approach handles the null-case straight off, which I think has a few benefits.

**Review by thermokarst — CHANGES_REQUESTED** [review]
Can we get a unit test to cover the case identified in #148?

File: q2_longitudinal/tests/test_longitudinal.py [hunk 2]
```diff
@@ -328,6 +328,17 @@ def test_pairwise_differences(self):
             individual_id_column='studyid', metric='observed_otus',
             replicate_handling='drop')
 
+    def test_pairwise_differences_zero_variance(self):
```
Comment (thermokarst) [inline q2_longitudinal/tests/test_longitudinal.py:331]: This test fails if the fix above is not in place. TDD!

**Review by thermokarst — APPROVED** [review]

**Comment by thermokarst** [comment]
@nbokulich, how does that test look to you?

**Comment by nbokulich** [comment]
LGTM thanks!
